src/pure/utils.ts

```diff
@@ -50,10 +50,19 @@ export function getVitestCommand(
     }
   }
 
-  const node_modules = path.resolve(projectRoot, 'node_modules')
   try {
-    if (!existsSync(node_modules))
+    const yarnLock = path.resolve(projectRoot, 'yarn.lock')
+    if (existsSync(yarnLock)) {
```

Comment:
To keep it simple, I just detect Yarn here and return "yarn vitest" right away. This might have side effects for projects running yarn and using node_modules and have a "vitest" script in their package.json. In a first thought, I think this might be the desired result anyways, so we do not need to find for the binary manually since we can do it through the package manager. However, if this is a problem, an alternative solution is to replace the two lines above with the solution below, which will only check for .pnp files. Thus, the projects with yarn and node_modules will fallback to the original implementation:

```
const hasPnpFiles = readdirSync(projectRoot).filter(file => /\.pnp\.(?:c|m)?js$/.test(file)).length > 0
if (hasPnpFiles) {
```

For this to work, we also need to import `readdirSync` at `import { existsSync, readdirSync } from 'fs-extra'`.

Please let me know if you would prefer this alternative.